miden-tx/src/host/account_procs.rs

```diff
@@ -17,23 +18,37 @@ impl AccountProcedureIndexMap {
     /// Merkle store of the provided advice provider.
     pub fn new<A: AdviceProvider>(account_code_root: Digest, adv_provider: &A) -> Self {
         // get the Merkle store with the procedure tree from the advice provider
-        let proc_store = adv_provider.get_store_subset([account_code_root].iter());
+        // let proc_store = adv_provider.get_store_subset([account_code_root].iter());
+
+        // get the account procedures from the advice_map
+        let procs = adv_provider.get_mapped_values(&account_code_root).unwrap();
 
         // iterate over all possible procedure indexes
         let mut result = BTreeMap::new();
-        for i in 0..AccountCode::MAX_NUM_PROCEDURES {
-            let index = NodeIndex::new(AccountCode::PROCEDURE_TREE_DEPTH, i as u64)
-                .expect("procedure tree index is valid");
-            // if the node at the current index does not exist, skip it and try the next node;this
-            // situation is valid if not all account procedures are loaded into the advice provider
-            if let Ok(proc_root) = proc_store.get_node(account_code_root, index) {
-                // if we got an empty digest, this means we got to the end of the procedure list
-                if proc_root == Digest::default() {
-                    break;
-                }
-                result.insert(proc_root, i as u8);
-            }
+
+        // for i in 0..AccountCode::MAX_NUM_PROCEDURES {
+        //     let index = NodeIndex::new(AccountCode::PROCEDURE_TREE_DEPTH, i as u64)
+        //         .expect("procedure tree index is valid");
+        //     // if the node at the current index does not exist, skip it and try the next node;this
+        //     // situation is valid if not all account procedures are loaded into the advice provider
+        //     if let Ok(proc_root) = proc_store.get_node(account_code_root, index) {
+        //         // if we got an empty digest, this means we got to the end of the procedure list
+        //         if proc_root == Digest::default() {
+        //             break;
+        //         }
+        //         result.insert(proc_root, i as u8);
+        //     }
+        // }
+        //
+
+        let mut i = 0;
```

Comment:
Here I have been able to extract procedures from initialized account code to populate the advice provider AccountProcedureIndexMap. 

I am filling the map with proc_root -> proc_elements K | V pairs.

This will enable us to query them from the internals of the virtual machine for procedure authentication.

Reply:
This looks right - though, I think you can probably simplify this as:
```rust
for (proc_idx, proc_info) in procs.chunks_exact(8).enumerate() {
    let root: [Felt; 4] = proc_info[0..4].try_into().expect("Slice with incorrect len.");
    result.insert(Digest::from(root), proc_idx);
}
```

Reply:
This does require that the advice map contains `account_code_commitment |-> account_procedure_elements` entry in it but inserting this entry seems to be commented out as I mentioned in one of the previous comments.

Reply:
Added it back in and cleaned the code.